crates/rpc/src/server/accept.rs

```diff
@@ -0,0 +1,188 @@
+use std::task::{Context as StdContext, Poll};
+
+use futures::{FutureExt, future::BoxFuture};
+use http::header::ACCEPT;
+use nom::{
+    IResult,
+    bytes::complete::{tag, take_until},
+};
+use semver::{Version, VersionReq};
+use tower::{Layer, Service};
+use tracing::debug;
+
+use crate::COMPONENT;
+
+/// Layer responsible for handling HTTP ACCEPT headers.
+#[derive(Clone)]
+pub struct AcceptLayer {
+    version: VersionReq,
+}
+
+impl AcceptLayer {
+    /// Create a new accept layer with the specified version requirement.
```

Comment:
nit: I think the phrasing here can be improved a bit: we are not specifying the version, we are reading from `Cargo.toml`.

Reply:
Yep, forgot to update the comment after updating impl